lighthouse-core/config/config.js

```diff
@@ -162,6 +161,48 @@ function assertValidGatherer(gathererInstance, gathererName) {
   }
 }
 
+
+/**
+ * Validate the LH.Flags
+ * @param {LH.Flags} flags
+ */
+function assertValidFlags(flags) {
+  // COMPAT: compatibility layer for devtools as it uses the old way and we need tests to pass
+  // TODO(paulirish): remove this from LH once emulation refactor has rolled into DevTools
+  // @ts-expect-error Deprecated flag
+  if (flags.channel === 'devtools' && flags.internalDisableDeviceScreenEmulation) {
+    // @ts-expect-error Deprecated flag
+    flags.formFactor = flags.emulatedFormFactor;
+    // @ts-expect-error Deprecated flag
+    flags.emulatedFormFactor = flags.internalDisableDeviceScreenEmulation = undefined;
+  }
+
+
+  // @ts-expect-error Checking for removed flags
+  if (flags.emulatedFormFactor || flags.internalDisableDeviceScreenEmulation) {
+    throw new Error('Invalid emulation flag. Emulation configuration changed in LH 7.0. See https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md');
+  }
+}
+
+/**
+ * Validate the settings after they've been built
+ * @param {LH.Config.Settings} settings
+ */
+function assertValidSettings(settings) {
+  if (!settings.formFactor) {
+    throw new Error(`\`settings.formFactor\` must be defined as 'mobile' or 'desktop'. See https://github.com/GoogleChrome/lighthouse/blob/master/docs/emulation.md`);
+  }
+
+  if (!settings.screenEmulation.disabled) {
+    // formFactor doesn't control emulation. So we don't want a mismatch:
+    //   Bad mismatch A: user wants mobile emulation but scoring is configured for desktop
+    //   Bad mismtach B: user wants everything desktop and set formFactor, but accidentally not screenEmulation
+    if (settings.screenEmulation.mobile !== (settings.formFactor === 'mobile')) {
```

Comment:
does this mean if you're testing on a real phone and you want to be graded like that, you need to do
```js
formFactor: 'mobile',
screenEmulation: {
  mobile: true,
  disabled: true,
  // ...
}
```
?

Reply:
I had the same thought but it's nested within a check on `disabled` so it should be fine just with `disabled`, right?

Reply:
> I had the same thought but it's nested within a check on `disabled` so it should be fine just with `disabled`, right?

Oops, yes, good catch. Maybe also worth calling out in the comment then?

Reply:
aside from the defaults you get for free, yes.
https://github.com/GoogleChrome/lighthouse/blob/31a7b554fc66201b86d6d6d32c473f053472f20e/docs/emulation.md#L19